Changed the API of the `SparseMerkleTree` to accept `MerkleStorageKey` (#485)
FuelLabs/fuel-vm: CHANGELOG.md

### Fixed

- [#482](https://github.com/FuelLabs/fuel-vm/pull/482): This PR address a security 
    issue where updates to a Sparse Merkle Tree could deliberately overwrite existing
    leaves by setting the leaf key to the hash of an existing leaf or node. This is 

>     done by removing the insertion of the leaf using the leaf key.

Reviewer: ```suggestion
    done by no longer inserting the leaf using the leaf key.
```
